src/module/entities/helper.js

```diff
@@ -1327,7 +1327,16 @@ module.exports = class UserProjectsHelper {
 						// Convert the names in 'validatedChildHierarchy' to strings and assign them to 'childHierarchyPath'
 						childHierarchyPath = validatedChildHierarchy.map(String)
 					}
-
+					if (singleEntity['targetedEntityTypes']) {
```

Comment:
@Prajwal17Tunerlabs Can we optimize this
singleEntity.targetedEntityTypes = Array.isArray(singleEntity.targetedEntityTypes)
	? await populateTargetedEntityTypesData(
			singleEntity.targetedEntityTypes.map(item => item.trim()),
			tenantId
	  )
	: []